### ISSUE TITLE
BUG: Set root does not work

### ISSUE THREAD

**Issue by henzigo** [artf/grapesjs#4083]
### GrapesJS version

- [X] I confirm to use the latest version of GrapesJS

### What browser are you using?

Safari 15.0

### Reproducible demo link

https://jsfiddle.net/adw6y12f/9/

### Describe the bug

Hello,
I'm trying to set root component for the editor as it was described [here](https://github.com/artf/grapesjs/pull/1185#issuecomment-395967321). Unfortunately, this is not working and I can edit all HTML content.

Thanks for your advice or fix that bug :)

**How to reproduce the bug?**
1. Create HTML with some divs
2. Set `id` or `class` to one div that you wants to be root layer
3. Set that `id` as root element to `init` method : 
```js
grapesjs.init({
  container: '#gjs',
  fromElement: 1,
  height: '100%',
  storageManager: { type: 0 },
  plugins: ['gjs-blocks-basic'],
  layerManager: {
    root: '#editable',
  },
});
```
4. You can edit all elements in editor

**What is the expected behavior?**
All elements will be rendered, but only elements in root element will be editable 

**What is the current behavior?**
All elements are editable

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by artf** [comment]
Yeah, I see the issue with the root configuration in LayerManager. The temporary fix would be to trigger the root change on render of the layers panel:
```js
editor.on('run:core:open-layers', () => editor.Layers.setRoot('#editable'));
```
But anyway, the root of the LayerManager doesn't decide if your components are editable or not, its purpose here is only to change the root in the LayerManager UI panel.

If you need to disable parts of your template, this is probably what you're looking for: https://jsfiddle.net/artur_arseniev/w6mvq4bx/

**Comment by henzigo** [comment]
Thank you for your answer. It would be nice to have some documentation of what is `root` element because only mentions are in PR/issues.

> If you need to disable parts of your template, this is probably what you're looking for: https://jsfiddle.net/artur_arseniev/w6mvq4bx/

Unfortunately, this is not exactly what I need. I will try to describe my situation:

I have using GrapesJS for editing blog articles. I want to make the use more enjoyable so before I initialize GrapesJS I download `template` of my frontend (This template is created dynamically, so it can be changed everytime I'm loading editor) and show this template in editor. Unfortunately I need to allow editing only one nested div, so I can't use your example. (The best way would be to invert logic you posted and allow editing only for div with specific class). 

Here is how my template page looks like on frontend
<img width="687" alt="Snímek obrazovky 2022-01-21 v 16 31 13" src="https://user-images.githubusercontent.com/1482966/150554768-a2b4859a-1b2c-4ed9-84a2-91ba9a200ea6.png">

After I'm done with styling article, I need save to my database only content of that nested div, so I can load contend via API on frontend.

Here is how my editor with loaded template looks like:
<img width="1458" alt="Snímek obrazovky 2022-01-21 v 16 17 53" src="https://user-images.githubusercontent.com/1482966/150552111-e632f6db-d2ab-456f-a451-853e87ce85a1.png">

Is there any way, how can I achieve my requirements at this time? Maybe in future, It would be nice to have some selector in configuration, that will do what I want. What do you think? 

Thank you for your advice ❤️

**Comment by artf** [comment]
Yeah, the LayerManager module itself has no documentation. Indeed it's on the waiting list for the refactoring and once it's done we can publish more about its usage and API documentation.

Anyway, I've seen your previous message about how you would like to use my demo example but with inverted logic (enabling only what is known). Well, I've tried it but unfortunately, I've spotted some issues I have to fix before. Once the next release is ready, I'll publish the demo example.

**Comment by artf** [comment]
Sorry @henzigo I forgot to update you: https://jsfiddle.net/artur_arseniev/aku70f5w/ (by using the latest version)

**Comment by henzigo** [comment]
Hello @artf

I appreciate that you posted demo for me :) It works quite well, but I have two issues with this code:

1. With my HTML page loaded CSS is not exported properly (There is my demo: https://jsfiddle.net/qjur4t3b/11/). Could it be, because of React and inline styles loaded? Is there any way how could I remove those CSS styles? 
2. I can still place components on top and bottom of my `editable` wrapper, but this is something I can live with

**Comment by artf** [comment]
> With my HTML page loaded CSS is not exported properly (There is my demo: https://jsfiddle.net/qjur4t3b/11/). Could it be, because of React and inline styles loaded? Is there any way how could I remove those CSS styles?

Yeah, you have to put those styles in an external file if you want them to be excluded from the export code. You can also try to update `getCss()` with the new option for the CSS code generator `editor.CodeManager.getCode(cmp, 'css', { onlyMatched: true })` to return only matched rules, but it will still show the rule if it's matched.

> I can still place components on top and bottom of my editable wrapper, but this is something I can live with

Yeah right, you can actually fix that by adding this to the wrapper:
```js
droppable: false,
propagate: ['highlightable', 'selectable', 'droppable']
```

**Comment by henzigo** [comment]
> Yeah, you have to put those styles in an external file if you want them to be excluded from the export code. You can also try to update getCss() with the new option for the CSS code generator editor.CodeManager.getCode(cmp, 'css', { onlyMatched: true }) to return only matched rules, but it will still show the rule if it's matched.

This is an unfortunate situation. We are using Styled components that are generating styles dynamically, so there is no chance to move those styles into separate files :(

**Comment by artf** [comment]
In that case, `onlyMatched: true` might help you, but consider that you're always able to filter CSS rules before exporting them so you can exclude what is not necessary for your output.
